client/components/language-picker/modal.jsx

```diff
@@ -417,7 +415,33 @@ export class LanguagePickerModal extends PureComponent {
 		this.handleClose();
 	};
 
-	handleClose = () => {
+	recordLanguagePickedEvent = ( isClosingWithoutSelection ) => {
+		const { selectedLanguageSlug, search } = this.state;
+
+		if (
+			isClosingWithoutSelection ||
+			this.initiallySelectedLanguageSlug === selectedLanguageSlug
+		) {
+			// we don't record an event if the language wasn't changed
+			return;
```

Comment:
We're making the assumption here that search only matters if we manage to set and change the language. How did we make that assertion? Can't the user find the search form useful even if they don't want to change their language right now? What if I, as a user, just want to know what languages are available at this time?

Reply:
That’s true, I took a shot in the dark with my assumption. If we just want to know how many users interacted with search regardless of whether it resulted in them selecting a new language then it is much simpler. Hopefully someone from i18n can chime in!

Reply:
I think it would be better to record the event no matter if it ends with actually switching the language or not. A few possible cases I can think of right now are if the user just wants to check for a language availability, or if the user fails to find a language, they would most likely end up not switching to another.